web/ui/react-app/src/pages/alerts/AlertContents.tsx

```diff
@@ -56,50 +80,90 @@ const AlertsContent: FC<AlertsProps> = ({ groups = [], statsCount }) => {
+  }, [groupList, filter]);
+
   return (
     <>
-      <div className="d-flex togglers-wrapper">
-        {stateColorTuples.map(([state, color]) => {
-          return (
-            <Checkbox
-              key={state}
-              wrapperStyles={{ marginRight: 20 }}
-              checked={filter[state]}
-              id={`${state}-toggler`}
-              onChange={toggleFilter(state)}
-            >
-              <Badge color={color} className="text-capitalize">
-                {state} ({statsCount[state]})
-              </Badge>
-            </Checkbox>
-          );
-        })}
-        <Checkbox
-          wrapperStyles={{ marginLeft: 'auto' }}
-          checked={showAnnotations.checked}
-          id="show-annotations-toggler"
-          onChange={({ target }) => setShowAnnotations({ checked: target.checked })}
-        >
-          <span style={{ fontSize: '0.9rem', lineHeight: 1.9 }}>Show annotations</span>
-        </Checkbox>
-      </div>
-      {groups.map((group, i) => {
-        const hasFilterOn = group.rules.some((rule) => filter[rule.state]);
-        return hasFilterOn ? (
-          <Fragment key={i}>
-            <GroupInfo rules={group.rules}>
-              {group.file} &gt; {group.name}
-            </GroupInfo>
-            {group.rules.map((rule, j) => {
-              return (
-                filter[rule.state] && (
-                  <CollapsibleAlertPanel key={rule.name + j} showAnnotations={showAnnotations.checked} rule={rule} />
-                )
-              );
-            })}
-          </Fragment>
-        ) : null;
-      })}
+      <Row className="align-items-center">
+        <Col className="d-flex">
+          {stateColorTuples.map(([state, color]) => {
+            return (
+              <Checkbox
+                key={state}
+                wrapperStyles={{ marginRight: 10 }}
+                checked={filter[state]}
+                id={`${state}-toggler`}
+                onChange={toggleFilter(state)}
+              >
+                <Badge color={color} className="text-capitalize">
+                  {state} ({statsCount[state]})
+                </Badge>
+              </Checkbox>
+            );
+          })}
+        </Col>
+        <Col xs="6">
+          <InputGroup>
+            <InputGroupAddon addonType="prepend">
+              <InputGroupText>{<FontAwesomeIcon icon={faSearch} />}</InputGroupText>
+            </InputGroupAddon>
+            <Input autoFocus onChange={handleSearchChange} placeholder="Filter by name or labels" />
+          </InputGroup>
+        </Col>
+        <Col className="d-flex flex-row-reverse">
+          <Checkbox
+            wrapperStyles={{ marginLeft: 'auto' }}
+            checked={showAnnotations.checked}
+            id="show-annotations-toggler"
+            onChange={({ target }) => setShowAnnotations({ checked: target.checked })}
+          >
+            <span style={{ fontSize: '0.9rem', lineHeight: 1.9 }}>Show annotations</span>
```

Comment:
Now at still pretty wide window widths, the "Show annotations" text is broken up over multiple lines, which doesn't look pretty:
![show-annotations-linebreak](https://user-images.githubusercontent.com/538008/149123353-f85f62b9-3ccc-432f-9b07-7ba60126ad0b.png)

I think for those screen widths, it would be better to already put the search input on its own line.

Reply:
do you mean whatever the screen size, the search bar is put in the next line ?

Reply:
I was more thinking to put it on its own line only when the screen width is small. But I haven't played around or thought about the best layout option for that yet :P

In any case, I don't think the "Show annotations" should ever have a line break.